nitransforms/manip.py

```diff
@@ -178,9 +177,8 @@ def asaffine(self, indices=None):
             The indices of the values to extract.
 
         """
-        affines = self.transforms if indices is None else np.take(self.transforms, indices)
-        retval = affines[0]
-        for xfm in affines[1:]:
+        retval = self.transforms[-1]
+        for xfm in reversed(self.transforms[:-1]):
             retval = xfm @ retval
         return retval
```

Comment:
I feel like it would be more intuitive to swap the arguments of the `@` than to reverse the order of the list:

```suggestion
        retval = affines[0]
        for xfm in affines[1:]:
            retval = retval @ xfm
        return retval
```

But we can also just use a reduce (I've added the imports above if you want to go this way):

```suggestion
        return reduce(op.matmul, self.transforms)
```